### PR TITLE
7zip: Add version 22.01 and Conan 2 compatibility

### PR DESCRIPTION
**Pull request by gjasny**
Specify library name and version:  **7zip/22.01**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/7zip/19.00/conanfile.py [hunk 1]
```diff
@@ -14,7 +18,8 @@ class SevenZipConan(ConanFile):
     license = ("LGPL-2.1", "BSD-3-Clause", "Unrar")
```
Comment (jwillikers) [inline recipes/7zip/19.00/conanfile.py:18]:
```suggestion
    license = ("LGPL-2.1-or-later", "BSD-3-Clause", "Unrar")
```

File: recipes/7zip/19.00/conanfile.py [hunk 2]
```diff
@@ -14,7 +18,8 @@ class SevenZipConan(ConanFile):
...
     homepage = "https://www.7-zip.org"
     topics = ("7zip", "zip", "compression", "decompression")
```
Comment (jwillikers) [inline recipes/7zip/19.00/conanfile.py:20]:
```suggestion
    topics = ("archive", "compression", "decompression", "zip")
```

File: recipes/7zip/19.00/conanfile.py [hunk 3]
```diff
@@ -55,47 +77,40 @@ def _msvc_platform(self):
...
+        if is_msvc(self):
             self._build_msvc()
         else:
             self._build_autotools()
 
     def package(self):
-        self.copy("DOC/License.txt", src="", dst="licenses")
-        self.copy("DOC/unRarLicense.txt", src="", dst="licenses")
+        copy(self, "License.txt", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder, "DOC"))
+        copy(self, "unRarLicense.txt", dst=os.path.join(self.package_folder, "licenses"), src=os.path.join(self.source_folder, "DOC"))
```
Comment (jwillikers) [inline recipes/7zip/19.00/conanfile.py:104]: Does this license only apply to the DLL? If we don't package the DLL, then maybe we don't need it?

**Review by jwillikers — CHANGES_REQUESTED** [review]
Just a question about the license really, otherwise it looks good.

File: recipes/7zip/19.00/conanfile.py (hunk 3 above)
Comment (gjasny) [reply recipes/7zip/19.00/conanfile.py:104]: No, it is explicitly mentioned in the `License.txt` and also relates to the source code:
```
  The licenses for files are:

    1) CPP/7zip/Compress/Rar* files: the "GNU LGPL" with "unRAR license restriction"
    2) CPP/7zip/Compress/LzfseDecoder.cpp: the "BSD 3-clause License"
    3) Some files are "public domain" files, if "public domain" status is stated in source file.
    4) the "GNU LGPL" for all other files. If there is no license information in 
       some source file, that file is under the "GNU LGPL".

  The "GNU LGPL" with "unRAR license restriction" means that you must follow both 
  "GNU LGPL" rules and "unRAR license restriction" rules.
```

And the generated binary contains rar support:
```
7z i

...
 0  ...F..................  Rar      rar r00       R a r ! 1A 07 00
 0  ...F..................  Rar5     rar r00       R a r ! 1A 07 01 00
...
```

Comment (jwillikers) [reply recipes/7zip/19.00/conanfile.py:104]: Thanks, I wasn't sure since their license web page says things differently: https://www.7-zip.org/license.txt

**Review by jwillikers — DISMISSED** [review]

File: recipes/7zip/config.yml [hunk 4]
```diff
@@ -1,3 +1,3 @@
 versions:
-  "19.00":
+  "22.01":
```
Comment (danimtb) [inline recipes/7zip/config.yml:2]: Why are you removing the 19.00 version? Normally we keep the versions that are already included in the repo unless we have many of them (in that case we keep only the latest patch).

Comment (gjasny) [reply recipes/7zip/config.yml:2]: The `19.00` version dependency `"lzma_sdk/9.20"` requires a Conan 2 port. I can port it as well if you think that the changes I did to 7zip look sane.

Comment (gjasny) [reply recipes/7zip/config.yml:2]: Let's wait for #17667

Comment (gjasny) [reply recipes/7zip/config.yml]: I re-added v19

**Review by jwillikers — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**